pkg/bindinfo/binding_cache.go

```diff
@@ -48,6 +48,101 @@ var GetBindingReturnNilAlways = stringutil.StringerStr("getBindingReturnNilAlway
 // LoadBindingNothing is only for test
 var LoadBindingNothing = stringutil.StringerStr("LoadBindingNothing")
 
+// digestBiMap represents a bidirectional map between noDBDigest and sqlDigest, used to support cross-db binding.
+// One noDBDigest can map to multiple sqlDigests, but one sqlDigest can only map to one noDBDigest.
+type digestBiMap interface {
+	// Add adds a pair of noDBDigest and sqlDigest.
+	// noDBDigest is the digest calculated after eliminating all DB names, e.g. `select * from test.t` -> `select * from t` -> noDBDigest.
+	// sqlDigest is the digest where all DB names are kept, e.g. `select * from test.t` -> exactDigest.
+	Add(noDBDigest, sqlDigest string)
+
+	// Del deletes the pair of noDBDigest and sqlDigest.
+	Del(sqlDigest string)
+
+	// NoDBDigest2SQLDigest converts noDBDigest to sqlDigest.
+	NoDBDigest2SQLDigest(noDBDigest string) []string
+
+	// SQLDigest2NoDBDigest converts sqlDigest to noDBDigest.
+	SQLDigest2NoDBDigest(sqlDigest string) string
+
+	// Copy copies this digestBiMap.
+	Copy() digestBiMap
+}
+
+type digestBiMapImpl struct {
+	mu                   sync.RWMutex
+	noDBDigest2SQLDigest map[string][]string // noDBDigest --> sqlDigests
+	sqlDigest2noDBDigest map[string]string   // sqlDigest --> noDBDigest
```

Comment:
sqlDigest2noDBDigest's value can use the ```[unique](https://pkg.go.dev/unique)``` to reduce memory

Reply:
OK I'll update here later, I have a few more refactoring PRs based on this, after merging them all I'll update here.

Reply:
unique handle seems interesting